src/main/java/net/wurstclient/hacks/AutoPlaceHack.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2014-2024 Wurst-Imperium and contributors.
+ *
+ * This source code is subject to the terms of the GNU General Public
+ * License, version 3. If a copy of the GPL was not distributed with this
+ * file, You can obtain one at: https://www.gnu.org/licenses/gpl-3.0.txt
+ */
+package net.wurstclient.hacks;
+
+import net.minecraft.client.gui.screen.ingame.HandledScreen;
+import net.minecraft.client.network.ClientPlayerEntity;
+import net.minecraft.item.BlockItem;
+import net.minecraft.util.ActionResult;
+import net.minecraft.util.Hand;
+import net.minecraft.util.hit.BlockHitResult;
+import net.minecraft.util.hit.HitResult;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Direction;
+import net.wurstclient.Category;
+import net.wurstclient.SearchTags;
+import net.wurstclient.events.UpdateListener;
+import net.wurstclient.hack.Hack;
+import net.wurstclient.settings.CheckboxSetting;
+import net.wurstclient.util.BlockUtils;
+
+@SearchTags({"bridge", "auto place"})
+public final class AutoPlaceHack extends Hack implements UpdateListener
+{
+	private final CheckboxSetting onlyBelowFeet =
+		new CheckboxSetting("Only Below Feet",
+			"Only place blocks when target block is 1 block below your feet.",
+			false);
+	private final CheckboxSetting fastPlace =
+		new CheckboxSetting("Always FastPlace",
+			"Builds as if FastPlace was enabled, even if it's not.", true);
+	
+	public AutoPlaceHack()
+	{
+		super("AutoPlace");
+		setCategory(Category.BLOCKS);
+		addSetting(onlyBelowFeet);
+		addSetting(fastPlace);
+	}
+	
+	@Override
+	public void onEnable()
+	{
+		EVENTS.add(UpdateListener.class, this);
+	}
+	
+	@Override
+	public void onDisable()
+	{
+		EVENTS.remove(UpdateListener.class, this);
+	}
+	
+	@Override
+	public void onUpdate()
+	{
+		if(MC.currentScreen instanceof HandledScreen)
+			return;
+		
+		if(MC.itemUseCooldown != 0 && !fastPlace.isChecked())
+			return;
+		
+		ClientPlayerEntity player = MC.player;
+		assert player != null;
+		
+		// Item in hand is not a block
+		if(!(player.getInventory().getStack(player.getInventory().selectedSlot)
+			.getItem() instanceof BlockItem))
+			return;
+		
+		HitResult hitResult = MC.crosshairTarget;
+		if(hitResult == null || hitResult.getType() != HitResult.Type.BLOCK)
+			return;
+		
+		BlockHitResult blockHitResult = (BlockHitResult)hitResult;
+		if(blockHitResult.getSide() == Direction.UP
+			|| blockHitResult.getSide() == Direction.DOWN)
+			return;
+		
+		if(!BlockUtils.canBeClicked(blockHitResult.getBlockPos()))
+			return;

```

Comment:
`BlockUtils.canBeClicked()` is not supposed to check for interactive blocks like chests and levers. It intentionally only checks if there is a voxel shape that you can click on, because the game would crash if you tried to click on a block with an empty shape.

Checking for interactivity would be useful in a lot of places, but sadly there isn't a clean way to do it. Minecraft only provides a click-on-it-and-see-what-happens type of interface. There is a primitive interactivity check in `AutoSoupHack` that checks against a list of known interactive blocks and entities. That check really should be renamed to `isInteractive()` and moved to a utility class one day, but that's out of scope for this PR.

Reply:
Sure, let's do that.